### PR TITLE
after setting up AG pyspecdata on windows

### PR DESCRIPTION
**Pull request by guinneak198**
JF set AG up with pyspecdata and this change was needed for running on windows

### CODE REVIEW

**Review by jmfranck — CHANGES_REQUESTED** [review]
Please see Slack messages… There should be changes to the documentation as well

**Comment by jmfranck** [comment]
This has been sitting around with a review, and no changes made

File: pyspecdata/mpl_utils.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import numpy as np
 from numpy import r_, pi
+from matplotlib import ticker
```
Comment (guinneak198) [inline pyspecdata/mpl_utils.py:3]: this was just a bug WK and I noticed when we tried to run a processing script we got an error upset about mticker - it didn't know what it was. I'm pretty certain this was a typo and this fix allows us to run scripts as usual

File: README.rst [hunk 2]
```diff
@@ -171,7 +170,7 @@ For reasons that we don't understand, the Fortran compiler can give odd errors,
 This appears to be Windows' fault, rather than conda's (?).
 We highly recommend trying both the Anaconda prompt, as well as the standard dos prompt (press start: type `cmd`) if you experience errors related to compilation.
 
-If you want to build the documentation, all run: `conda install -y -c conda-forge sphinx_rtd_theme sphinx-gallery`
+If you want to build the documentation, run: `conda install -y -c conda-forge sphinx_rtd_theme sphinx-gallery`
```
Comment (guinneak198) [inline README.rst:173]: I don't think we want the "all" run here? but correct me if I'm wrong

File: README.rst [hunk 3]
```diff
@@ -159,10 +159,9 @@ Rather, you can just import ``mayavi.mlab`` and pass it to any figure list that
 
 Installation for developers
 ---------------------------
-
-Once these are installed,
-to install from github, just ``git clone https://github.com/jmfranck/pyspecdata.git`` then move to the directory where setup.py lives,
-and do
+Typically, we create a folder on the hard drive called ``git`` or ``git_repos``, where you will put all of your code and the pyspecdata folder that you are about to clone. After installing the packages via the anaconda prompt, switch to your git bash terminal and move into your ``git`` folder.  
+To install from github, just ``git clone https://github.com/jmfranck/pyspecdata.git``. Then switch over to the anaconda prompt and move to the directory where setup.py lives (ex. ``git/pyspecdata/``),
```
Comment (guinneak198) [inline README.rst:163]: we decided it was best to just run setup.py on anaconda but git clone is a git bash command I'm pretty sure so we have to clone in git bash and then move back to anaconda to run setup.py right?

File: README.rst [hunk 4]
```diff
@@ -110,7 +110,7 @@ in our hands -- we are happy to hear from you and work with you to try to
 broaden its applicability!
 
 On **Windows** with `Anaconda 3.X <https://www.anaconda.com/blog/individual-edition-2020-11>`_,
-just run ``conda install -y -c anaconda numpy scipy sympy pyqt pytables matplotlib h5py libpython`` followed by ``conda install -y m2w64-toolchain`` (the libpython and m2w64-toolchain are for building compiled extensions such as the ILT).
+just run ``conda install -y -c anaconda numpy scipy sympy pyqt pytables matplotlib h5py libpython pylab itertools collections lmfit`` followed by ``conda install -y m2w64-toolchain`` (the libpython and m2w64-toolchain are for building compiled extensions such as the ILT).
```
Comment (guinneak198) [inline README.rst:113]: extra packages needed to run the examples in pyspecdata/examples

Comment (jmfranck) [reply README.rst:113]: Aren't iter tools and collections part of the base python installation?

File: README.rst (hunk 3 above)
Comment (jmfranck) [reply README.rst:163]: The business about the directories is not necessary... People can organize their files however they like. On the other hand, I would explicitly say that it's best not to run setup.py inside the bash terminal.

File: pyspecdata/mpl_utils.py (hunk 1 above)
Comment (jmfranck) [inline pyspecdata/mpl_utils.py:3]: Import ticker as mticker, and then do not make the change that you make below

File: pyspecdata/datadir.py [hunk 5]
```diff
@@ -371,6 +371,7 @@ def rclone_search(fname,exp_type,dirname):
                 remotelocation,
                 # dirname below needs to be replaced with path relative to current directory
                 os.path.normpath(os.path.join(dirname)).replace('\\','\\\\')))
+    cmd = cmd.replace("'",'"')
```
Comment (jmfranck) [inline pyspecdata/datadir.py:374]: You also want it to be clear how to use the rclone thing. There is already some documentation under search file name and find file that you could point to from the main readme file or from index.rst, as makes sense to you. 

But a big improvement that you can make here is that now you've been through the process of doing the rclone setup, would you be able to do that based on the documentation as it is now?

You can alsep search the directory for rclone to find existing documentation

**Review by jmfranck — CHANGES_REQUESTED** [review]
Overall, this was pretty good, and I liked the explanatory comments. Please just look at my comments and responses to resolve

File: README.rst (hunk 4 above)
Comment (jmfranck) [reply README.rst:113]: this comment has not been addressed

File: README.rst (hunk 3 above)
Comment (jmfranck) [reply README.rst:163]: this comment has not been addressed

File: pyspecdata/mpl_utils.py (hunk 1 above)
Comment (jmfranck) [reply pyspecdata/mpl_utils.py:3]: this comment has not been addressed

File: pyspecdata/mpl_utils.py [hunk 6]
```diff
@@ -8,8 +9,8 @@ def gridandtick(ax,rotation=(0,0),precision=(2,2),
         formatonly = False,fixed_y_locator = None,
         use_grid = True,
         spines = None,y = True):
-    defaultMajorLocator = lambda: mticker.MaxNLocator(min_n_ticks=4, steps=[1,2,5,10])
-    defaultMinorLocator = lambda: mticker.AutoMinorLocator(n=5)
+    defaultMajorLocator = lambda: ticker.MaxNLocator(min_n_ticks=4, steps=[1,2,5,10])
+    defaultMinorLocator = lambda: ticker.AutoMinorLocator(n=5)
```
Comment (jmfranck) [inline pyspecdata/mpl_utils.py:13]: this is what I refer to in the comment above, which has not been addressed

**Review by jmfranck — CHANGES_REQUESTED** [review]

File: pyspecdata/mpl_utils.py (hunk 1 above)
Comment (guinneak198) [reply pyspecdata/mpl_utils.py:3]: edited

File: README.rst [hunk 7]
```diff
@@ -159,10 +159,8 @@ Rather, you can just import ``mayavi.mlab`` and pass it to any figure list that
 
 Installation for developers
 ---------------------------
-
-Once these are installed,
-to install from github, just ``git clone https://github.com/jmfranck/pyspecdata.git`` then move to the directory where setup.py lives,
-and do
+To install from github, just ``git clone https://github.com/jmfranck/pyspecdata.git``. Then switch over to the anaconda prompt and move to the directory where setup.py lives (ex. ``git/pyspecdata/``),
```
Comment (jmfranck) [inline README.rst:162]: what about making sure `conda info` matches, though?

File: pyspecdata/datadir.py (hunk 5 above)
Comment (jmfranck) [reply pyspecdata/datadir.py:374]: I don't think that this has been addressed

**Review by jmfranck — CHANGES_REQUESTED** [review]
Replied and resolved the conversations that have been taken care of.

File: README.rst (hunk 7 above)
Comment (guinneak198) [reply README.rst:162]: README.rst is for installing pyspecdata. I've added the conda info in the step where we install anaconda in the wiki (https://avhe-jmfranck-node1.syr.edu/wiki/index.php/Important_Links#GVim_and_PySpecData). The installation of pyspecdata assumes this is already done properly, but will add a note

Comment (jmfranck) [reply README.rst:162]: but `conda info` is relevant to installing pyspecdata as a developer -- ostensibly if you're doing that, you have git bash installed

Comment (jmfranck) [reply README.rst:162]: it should be possible to just refer the wiki to the readme -- say do the developer install

File: README.rst [hunk 8]
```diff
@@ -109,8 +109,8 @@ In all situations, note that this is a development library that works very well
 in our hands -- we are happy to hear from you and work with you to try to
 broaden its applicability!
 
-On **Windows** with `Anaconda 3.X <https://www.anaconda.com/blog/individual-edition-2020-11>`_,
-just run ``conda install -y -c anaconda numpy scipy sympy pyqt pytables matplotlib h5py libpython`` followed by ``conda install -y m2w64-toolchain`` (the libpython and m2w64-toolchain are for building compiled extensions such as the ILT).
+On **Windows**, you must first check that the output of ``conda info`` on your git bash matches the output of your anaconda prompt. If these do not match you will have to make the conda environment accessible to all users. Once this is checked, with `Anaconda 3.X <https://www.anaconda.com/blog/individual-edition-2020-11>`_,
+just run ``conda install -y -c anaconda numpy scipy sympy pyqt pytables matplotlib h5py libpython pylab lmfit`` followed by ``conda install -y m2w64-toolchain`` (the libpython and m2w64-toolchain are for building compiled extensions such as the ILT).
```
Comment (guinneak198) [inline README.rst:113]: note on conda info - thought this was the best place but what's your take?

File: pyspecdata/datadir.py (hunk 5 above)
Comment (guinneak198) [reply pyspecdata/datadir.py:374]: added rclone note to README.rst

File: README.rst [hunk 9]
```diff
@@ -110,7 +110,7 @@ in our hands -- we are happy to hear from you and work with you to try to
 broaden its applicability!
 
 On **Windows** with `Anaconda 3.X <https://www.anaconda.com/blog/individual-edition-2020-11>`_,
-just run ``conda install -y -c anaconda numpy scipy sympy pyqt pytables matplotlib h5py libpython pylab lmfit`` followed by ``conda install -y m2w64-toolchain`` (the libpython and m2w64-toolchain are for building compiled extensions such as the ILT).
+just run ``conda install -y -c anaconda numpy scipy sympy pyqt pytables matplotlib h5py libpython`` followed by ``conda install -y m2w64-toolchain`` (the libpython and m2w64-toolchain are for building compiled extensions such as the ILT).
```
Comment (jmfranck) [inline README.rst:113]: why did you remove lmfit?

File: README.rst [hunk 10]
```diff
@@ -148,6 +148,10 @@ that you install the following packages using a good package-management system (
 
 * h5py
 
+* pylab
```
Comment (jmfranck) [inline README.rst:151]: I'm a little confused -- I think that pylab is typically part of matplotlib

File: README.rst [hunk 11]
```diff
@@ -159,7 +163,9 @@ Rather, you can just import ``mayavi.mlab`` and pass it to any figure list that
 
 Installation for developers
 ---------------------------
-To install from github, just ``git clone https://github.com/jmfranck/pyspecdata.git``. Then switch over to the anaconda prompt and move to the directory where setup.py lives (ex. ``git/pyspecdata/``),
+**Warning** Before running the installation for developers, you must first check that the output of ``conda info`` on your git bash terminal matches the output of your anaconda prompt. If these do not match you will have to make the conda environment accessible to all users. 
+
+Once this is checked, to install pySpecData from github, just ``git clone https://github.com/jmfranck/pyspecdata.git``. Then switch over to the anaconda prompt and move to the directory where setup.py lives (ex. ``git/pyspecdata/``),
```
Comment (jmfranck) [inline README.rst:168]: this is called the "root directory of the repository"

Comment (jmfranck) [reply README.rst:168]: don't give this example -- no guarantee that other people organize their personal directory like you

File: README.rst [hunk 12]
```diff
@@ -159,7 +163,9 @@ Rather, you can just import ``mayavi.mlab`` and pass it to any figure list that
 
 Installation for developers
 ---------------------------
-To install from github, just ``git clone https://github.com/jmfranck/pyspecdata.git``. Then switch over to the anaconda prompt and move to the directory where setup.py lives (ex. ``git/pyspecdata/``),
+**Warning** Before running the installation for developers, you must first check that the output of ``conda info`` on your git bash terminal matches the output of your anaconda prompt. If these do not match you will have to make the conda environment accessible to all users.
```
Comment (jmfranck) [inline README.rst:166]: remove second sentence

File: README.rst [hunk 13]
```diff
@@ -171,6 +177,27 @@ We highly recommend trying both the Anaconda prompt, as well as the standard dos
...
+The value should correspond either to your Rclone remote or you can simply copy and
+paste the location of your data on the machine.
+
+Rclone is a useful tool to get files off of the team's drive or whatever cloud storage you may use. 
+This enables you to store datafiles locally allowing the user to work offline if needed.
+Prior to setting up Rclone you will need to make a local file (in your home directory) 
+where your data files will be added to as called. Typically, the name of this folder corresponds
+to the name of the folder on the cloud (e.g. 'exp_data').
+
+To get set up with Rclone, download Rclone and follow the documentation which should include
```
Comment (jmfranck) [inline README.rst:195]: this should be a subsection of above -- look up an rst cheatsheet for how to look at levels of secitons

File: README.rst [hunk 14]
```diff
@@ -171,8 +175,29 @@ For reasons that we don't understand, the Fortran compiler can give odd errors,
...
+If you want to build the documentation, run: `conda install -y -c conda-forge sphinx_rtd_theme sphinx-gallery`
+
+Setting up your _pyspecdata configuration file
+----------------------------------------------
+Part of the pySpecData package is the datadir module, allowing the user to run the same code on 
+different machines - even thought he location of the raw spectral data might change. 
+This is controlled by the ``~/.pyspecdata`` or ``~/_pyspecdata`` config file. 
+in the key ``[General]`` there should be a value pair for ``data_directory = ``.
+The value should correspond either to your Rclone remote or you can simply copy and
+paste the location of your data on the machine.
```
Comment (jmfranck) [inline README.rst:187]: I would give this as an example (use a [literal block](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks) to format)

```
[General]
data_directory = /home/jmfranck/exp_data
qesr conversion = 162.66
default q = 4700

[ExpTypes]
odnp_nmr_comp/odnp = /home/jmfranck/exp_data/NMR_comp/ODNP

[mode]
figures = standard

[RcloneRemotes]
nmr_comp/odnp = jmf_teams:General/exp_data/NMR_comp/ODNP/
```
I would use a bulleted list to explain the file in order.

Work in this phrase: "where `jmf_teams` is a properly configured (rclone)[https://rclone.org/] remote that shows up in response to the shell command `rclone config`"

**Review by jmfranck — CHANGES_REQUESTED** [review]

File: README.rst (hunk 10 above)
Comment (guinneak198) [reply README.rst:151]: good catch forgot I took this out above

File: README.rst [hunk 15]
```diff
@@ -195,9 +187,50 @@ to the name of the folder on the cloud (e.g. 'exp_data').
...
+
+    [ExpTypes]
+    odnp_nmr_comp/odnp = /home/jmfranck/exp_data/NMR_comp/ODNP
+
+gives the location of the ODNP datasets. So when you call ``odnp_nmr_comp/odnp`` this will point
+to the actual location, ``/home/jmfranck/exp_data/NMR_comp/ODNP``
+
+Finally add in the ``RcloneRemote`` key to your config file:
+
+::
```
Comment (jmfranck) [inline README.rst:224]: please gather the file into a single file, and then explain underneath.  It's easy to refer to the sections

**Review by jmfranck — CHANGES_REQUESTED** [review]

File: README.rst (hunk 15 above)
Comment (jmfranck) [reply README.rst:232]: most of these sections are created automatically -- you're not telling them how to write a config file from scratch -- just what it means, so that they can edit it

File: README.rst (hunk 7 above)
Comment (guinneak198) [reply README.rst]: wiki already points to this guy for installing pyspecdata so I think we are good?

File: README.rst (hunk 15 above)
Comment (guinneak198) [reply README.rst]: okay condensed literal block and discuss what each section is
